src/UserProfileService/Controllers/V2/UsersController.cs

```diff
@@ -208,18 +206,28 @@ await _readService.GetProfilesAsync<UserBasic, GroupBasic, OrganizationBasic>(
     public async Task<IActionResult> GetOwnProfileAsync(CancellationToken cancellationToken = default)
     {
         _logger.EnterMethod();
-        string currentUserId = await _userContextStore.GetIdOfCurrentUserAsync();
 
-        var profile =
-            await _readService.GetProfileAsync<User>(
+        string currentUserId = _userContextStore.GetIdOfCurrentUser();
+
+        List<IProfile> profiles =
+            await _readService.GetProfilesByExternalOrInternalIdAsync<User, Group, Organization>(
                 currentUserId,
-                RequestedProfileKind.User,
                 cancellationToken: cancellationToken);
 
-        IActionResult result =
-            ActionResultHelper.ToActionResult(profile.ResolveUrlProperties(ControllerContext));
+        IProfile profile = profiles.FirstOrDefault();
```

Comment:
You could combine your `FirstOrDefault(...)` with your `if (... || profile is not { Kind: ProfileKind.User }` and check if the resulting profile is `null`.
IMHO it is more clean.